src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py

```diff
@@ -0,0 +1,20 @@
+import pandas as pd
+from vivarium import Artifact
+
+from vivarium_census_prl_synth_pop.constants import data_keys, paths
+
+
+def get_nicknames_proportion():
+    # This function calculates the proportion of names that have potential nicknames. This will return a float that
+    # that will be used as a constant to scale the nicknames noise function in the pseudopeople packages.
+
+    art = Artifact(paths.DEFAULT_ARTIFACT)
+    names = art.load(data_keys.SYNTHETIC_DATA.FIRST_NAMES).reset_index()["name"]
+    # Load and format nicknames dataset
+    names_with_nicknames = pd.read_csv(paths.NICKNAMES_DATA_PATH)
+    names_with_nicknames = names_with_nicknames["name"].apply(lambda x: x.title())
+
+    # Proportion
+    nickname_eligible = names.isin(names_with_nicknames)
```

Comment:
Do you know that all the names in the artifact are title case?

Reply:
Yes

Reply:
Are you sure? There are a lot of African American names like DeAngelo, LaToya etc. that aren't title case. I suspect those names aren't in our nicknames csv though, so it probably doesn't matter.

Reply:
Yeah, I was assuming we would do a case-insensitive match to determine whether a name is eligible for a nickname, e.g. convert all the simulation names to lower case to match the nicknames CSV. And yeah, converting all the chosen nicknames to title case seems reasonable.